Buildings/Applications/DHC/EnergyTransferStations/CoolingIndirect.mo

```diff
@@ -0,0 +1,316 @@
+
+  Modelica.Blocks.Interfaces.RealInput TSet "Setpoint temperature"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
+
+  Buildings.Fluid.HeatExchangers.PlateHeatExchangerEffectivenessNTU hex(
+    redeclare final package Medium1 = Medium,
+    redeclare final package Medium2 = Medium,
+    final m1_flow_nominal=m1_flow_nominal,
+    final m2_flow_nominal=m2_flow_nominal,
+    final dp1_nominal=dp1_nominal,
+    final dp2_nominal=dp2_nominal,
+    final configuration=Buildings.Fluid.Types.HeatExchangerConfiguration.CounterFlow,
+    final use_Q_flow_nominal=true,
+    final Q_flow_nominal=Q_flow_nominal,
+    final T_a1_nominal=T_a1_nominal,
+    final T_a2_nominal=T_a2_nominal) "Indirect cooling heat exchanger"
+    annotation (Placement(transformation(extent={{20,-10},{40,10}})));
+
+  Buildings.Controls.Continuous.LimPID con(
+    final controllerType=Modelica.Blocks.Types.SimpleController.PID,
+    final k=k,
+    final Td=Td,
+    final yMax=yMax,
+    final yMin=yMin,
+    final Ti=Ti,
+    final wp=wp,
+    final wd=wd,
+    final Ni=Ni,
+    final Nd=Nd,
+    final initType=Modelica.Blocks.Types.InitPID.InitialOutput,
+    final xi_start=xi_start,
+    final xd_start=xd_start,
+    final y_start=yCon_start,
+    final reverseAction=reverseAction) "Controller"
+    annotation (Placement(transformation(extent={{-90,-10},{-70,10}})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTDisSup(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=m1_flow_nominal)
+    "District-side (primary) supply temperature sensor"
+    annotation (Placement(transformation(extent={{-90,50},{-70,70}})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTDisRet(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=m1_flow_nominal)
+    "District-side (primary) return temperature sensor"
+    annotation (Placement(transformation(extent={{70,50},{90,70}})));
+
+  Modelica.Blocks.Continuous.Integrator int(k=1) "Integration"
+    annotation (Placement(transformation(extent={{60,120},{80,100}})));
+
+  Buildings.Fluid.Sensors.MassFlowRate senMasFlo(redeclare package Medium = Medium)
+    annotation (Placement(transformation(extent={{-60,50},{-40,70}})));
+
+  Buildings.Fluid.Sensors.TemperatureTwoPort TBuiRet(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=m2_flow_nominal)
+    "Building-side (secondary) return temperature"
+    annotation (Placement(transformation(extent={{-70,-70},{-90,-50}})));
+
+  Buildings.Fluid.Actuators.Valves.TwoWayQuickOpening val(
+    redeclare final package Medium = Medium,
+    final m_flow_nominal=m1_flow_nominal,
+    final dpValve_nominal=dpValve_nominal,
+    riseTime(displayUnit="s") = 60,
+    y_start=0) "District-side (primary) control valve"
+    annotation (Placement(transformation(extent={{-30,70},{-10,50}})));
+
+  Modelica.Blocks.Math.Gain cp(k=cp_default)
```

Comment:
it can be finalized.
`final k=cp_default`.

Reply:
Done